velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -634,12 +634,22 @@ namespace {
 bool applyPartitionFilter(
     const TypePtr& type,
     const std::string& partitionValue,
+    const std::unordered_map<std::string, std::string>& columnParameters,
     common::Filter* filter) {
   if (type->isDate()) {
-    const auto result = util::fromDateString(
-        StringView(partitionValue), util::ParseMode::kPrestoCast);
-    VELOX_CHECK(!result.hasError());
-    return applyFilter(*filter, result.value());
+    int32_t result = 0;
+    // Iceberg partition values are already in daysSinceEpoch, no need to
+    // convert.
+    if (columnParameters.count(
```

Comment:
This check is repeated twice. Can we abstract this check as a method of ColumnHandle ?